es_managers/es_idx_manager/app.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+"""
+This AWS Lambda function allowed to delete the old Elasticsearch index
+"""
+from botocore.auth import SigV4Auth
+from botocore.awsrequest import AWSRequest
+from botocore.credentials import create_credential_resolver
+from botocore.session import get_session
+from botocore.vendored.requests import Session
+import sys
+if sys.version_info[0] == 3:
+    from urllib.request import quote
+else:
+    from urllib import quote
+import datetime
+import json
+import time
+import os
+import yaml
+
+import domovoi
+
+class ES_Exception(Exception):
+    """Exception capturing status_code from Client Request"""
+    status_code = 0
+    payload = ""
+
+    def __init__(self, status_code, payload):
+        self.status_code = status_code
+        self.payload = payload
+        Exception.__init__(self,
+                           "ES_Exception: status_code={}, payload={}".format(
+                               status_code, payload))
+
+
+class ES_Cleanup(object):
+
+    name = "lambda_es_cleanup"
+
+    def __init__(self, event, context, cluster_config):
+        """Main Class init
+       
+        Args:
+            event (dict): AWS Cloudwatch Scheduled Event
+            context (object): AWS running context
+        """
+        self.report = []
+        self.event = event
+        self.context = context
+
+        self.cfg = {}
+        self.cfg["es_endpoint"] = cluster_config.get('endpoint')
+        self.cfg["index"] = [index.get("prefix") for index in cluster_config.get('indices')]
+
+        self.cfg["delete_after"] = cluster_config.get('days')
+        self.cfg["es_max_retry"] = 3
+        self.cfg["index_format"] = cluster_config.get('index_format')
+        self.cfg["sns_alert"] = cluster_config.get('sns_alert', "")
+
+        if not self.cfg["es_endpoint"]:
+            raise Exception("[es_endpoint] OS variable is not set")
+
+    def send_to_es(self, path, method="GET", payload={}):
+        """Low-level POST data to Amazon Elasticsearch Service generating a Sigv4 signed request
+        
+        Args:
+            path (str): path to send to ES
+            method (str, optional): HTTP method default:GET
+            payload (dict, optional): additional payload used during POST or PUT
+        
+        Returns:
+            dict: json answer converted in dict
+        
+        Raises:
+            #: Error during ES communication
+            ES_Exception: Description
+        """
+        if not path.startswith("/"):
+            path = "/" + path
+
+        es_region = self.cfg["es_endpoint"].split(".")[1]
+
+        # send to ES with exponential backoff
+        retries = 0
```

Comment:
To make your code a little simpler, you might consider using a [`@retry` decorator](https://pypi.python.org/pypi/retrying). No worries, if this seems like too much.